shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -760,6 +764,47 @@ - (void)showAutocorrectionPromptRectForStart:(NSUInteger)start
 
 #pragma mark - FlutterViewEngineDelegate
 
+- (void)showToolbar:(int)client {
+  [_textInputChannel.get() invokeMethod:@"TextInputClient.showToolbar" arguments:@[ @(client) ]];
+}
+
+- (void)focusElement:(UIScribbleElementIdentifier)elementIdentifier
+             atPoint:(CGPoint)referencePoint
+              result:(id)callback {
+  [_textInputChannel.get()
+      invokeMethod:@"TextInputClient.focusElement"
+         arguments:@[ elementIdentifier, @(referencePoint.x), @(referencePoint.y) ]
+            result:callback];
+}
+
+- (void)requestElementsInRect:(CGRect)rect result:(id)callback {
+  [_textInputChannel.get()
+      invokeMethod:@"TextInputClient.requestElementsInRect"
+         arguments:@[ @(rect.origin.x), @(rect.origin.y), @(rect.size.width), @(rect.size.height) ]
+            result:callback];
+}
+
+- (void)scribbleInteractionBegan {
+  [_textInputChannel.get() invokeMethod:@"TextInputClient.scribbleInteractionBegan" arguments:nil];
```

Comment:
should there be a framework pr associated with the change? it may be easier to review if you can also create the framework part pr

Reply:
Yes, I should have put a link to it in the description. Here it is: https://github.com/flutter/flutter/pull/75472